src/SAML2/XML/spid/Issuer.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+declare(strict_types=1);
+
+namespace SAML2\XML\spid;
+
+use DOMElement;
+use SAML2\Constants;
+use SAML2\XML\saml\NameIDType;
+use Webmozart\Assert\Assert;
+
+/**
+ * Class representing the saml:Issuer element compliant with SPID spefication.
+ *
+ * @author Tim van Dijen, <tvdijen@gmail.com>
+ * @package simplesamlphp/saml2
+ */
+final class Issuer extends NameIDType
```

Comment:
Does this element use the SAML metadata namespace, according to the SPID specification? I'm wondering if that's allowed, because they would essentially be overriding the original schema.

If that's the case, I wonder as well how can we process these elements if we receive an XML with a regular `md:Issuer` element that follows the SPID specification instead of the SAML standard....

Reply:
They just follow the namespace for SAML :(

Reply:
Maybe we should not enforce too much of the spec in the library and only focus on XML structural integrity and presence of mandatory fields..
Business-logic like this can then go into SSP, together with SAML2int rules..

Reply:
There's a third approach for cases like this, which is completely ignoring the issue during parsing. As in: "_there shouldn't be this field here, let's just ignore it then_". Now, that would work on the IdP side. We could essentially ignore the attribute to the point of not even setting the property internal to the `Issuer` object. That way a SimpleSAMLphp will at least not choke with an authentication request coming from an SPID service provider. If you really want to get the `NameQualifier` attribute (which you shouldn't be using), well, you can always fetch it manually.

Now, from the SP point of view, I think it's important that messages produced by our library don't break the standard. I can live with this being done in a separate class, because then it's the responsibility of the service to break it consciously. In any case, the SPID implementation of `Issuer` can be greatly simplified, because the format is fixed, and only the `NameQualifier` and the value are required.

There's a third case, which is an IdP sending a response with an assertion. Assertions are fine, but if we look at the example response, the `Issuer` there contains both `Format` and `NameQualifier`. However, the text describing the `Issuer` element of a response doesn't say anything about those:

> the element &lt;Issuer&gt;must be present, indicating the entityID of the issuing entity, i.e. the Identity Provider itself; the attribute format must be omitted or have the value “urn:oasis:names:tc:SAML:2.0:nameid-format:entity”;

I'm therefore inclined to ignore the example and assume this was a mistake by the ones writing the spec. That would leave us with the custom, very basic class for authentication requests, and a relaxed test for everything else, ignoring any extra attributes if the format is _entity_. Would that be reasonable?

Reply:
Not sure I follow.. Would it be to much to ask to just do add your last suggestion to this PR? It sound like a minor change.

Reply:
My response was way more convoluted than the code I'd suggest 🤣

I'll do it 👍

Reply:
There, have a look!